llvm/lib/Transforms/Utils/InlineFunction.cpp

```diff
@@ -2795,13 +2802,10 @@ llvm::InlineResult llvm::InlineFunction(CallBase &CB, InlineFunctionInfo &IFI,
   }
 
   if (ConvergenceControlToken) {
-    auto *I = FirstNewBlock->getFirstNonPHI();
-    if (auto *IntrinsicCall = dyn_cast<IntrinsicInst>(I)) {
-      if (IntrinsicCall->getIntrinsicID() ==
-          Intrinsic::experimental_convergence_entry) {
-        IntrinsicCall->replaceAllUsesWith(ConvergenceControlToken);
-        IntrinsicCall->eraseFromParent();
-      }
+    auto *IntrinsicCall = getConvergenceEntry(*FirstNewBlock);
```

Comment:
The coding style encourages using the actual type name in a declaration instead of `auto`. This is mainly for readability when the type is not obvious from context. For example, `auto *D = dyn_cast<DerivedClass>(B)` is fine because it's obvious that the type of `D` is `DerivedClass *`.